sql/analyze_test.go

```diff
@@ -358,7 +360,7 @@ func TestSimplifyAndExprCheck(t *testing.T) {
 			t.Errorf("%s: expected %s, but found %s", d.expr, d.expected, s)
 		}
 
-		if d.check && !checkEquivExpr(t, expr1, expr2, qvals) {
+		if d.checkEquiv && !checkEquivExpr(t, expr1, expr2, qvals) {
```

Comment:
I think `checkEquivExpr` should return an error, which will make this handling way clearer